src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1314,19 +1308,13 @@ internal virtual HttpResponseMessage GetResponse(HttpClient client, HttpRequestM
                     {
                         FillRequestStream(requestWithoutRange);
 
-                        long requestContentLength = requestWithoutRange.Content is null ? 0 : requestWithoutRange.Content.Headers.ContentLength.Value;
+                        response.Dispose();
+                        response = GetResponse(client, requestWithoutRange, handleRedirect);
 
-                        string reqVerboseMsg = string.Format(
-                            CultureInfo.CurrentCulture,
-                            WebCmdletStrings.WebMethodInvocationVerboseMsg,
-                            requestWithoutRange.Version,
-                            requestWithoutRange.Method,
-                            requestContentLength);
+                        long requestContentLength = requestWithoutRange.Content is null ? 0 : requestWithoutRange.Content.Headers.ContentLength.Value;
+                        string reqVerboseMsg = string.Format(CultureInfo.CurrentCulture, WebCmdletStrings.WebMethodInvocationVerboseMsg, response.Version, requestWithoutRange.Method, requestContentLength);
```

Comment:
if we're getting the version from the response, why aren't we getting the content length from the response?

Reply:
In this PR I'm changing 3 verbose messages: 2 `reqVerboseMsg` and 1 `resVerboseMsg`.
```ps1
$x = Invoke-WebRequest https://httpbin.org/post -Method post -Verbose -Body @{"A"="B"} -HttpVersion 3
#VERBOSE: HTTP/3.0 POST with 3-byte payload <-- `reqVerboseMsg` 
#VERBOSE: received 471-byte response of content type application/json <-- `resVerboseMsg`
```
`reqVerboseMsg` --> { requested `HttpVersion` (it's imprecise, my pc does not support HTTP/3.0) }  { `HttpMethod` }  { request `ContentLength` from -Body }

`resVerboseMsg` --> { response `ContentLength` }  { `content type` }

PROPOSED SOLUTION
`reqVerboseMsg` --> { response `HttpVersion` }

BETTER SOLUTION
VERBOSE: HTTP/3.0 POST with 3-byte payload
VERBOSE: HTTP/2.0 received 471-byte response of content type application/json

We never connect to the remote server with HTTP/3.0
Test:
```ps1
$x.BaseResponse.Version
<#
Major  Minor  Build  Revision
-----  -----  -----  --------
2      0      -1     -1
#>
```
because of this @mklement0 opened an issue asking to investigate.
I checked curl behaviour and observed that it never writes HTTP/3.0 if its not supported, so i opened this pr to change our behaviour to match curl

Reply:
I think we want to have clear request and response verbose messages.  In fact, I would even suggest that we have separate resource strings indicating `Request: ...` and `Response: ...`.